src/cdrizzlebox.c

```diff
@@ -80,172 +82,180 @@ compute_bit_value(integer_t uuid) {
+                            area += 0.5 * width *
+                                    ((1.0 + delta[0]) + (1.0 + delta[1]));
+                        }
+                    }
+
+                } else {
+                    /* If ends of the line segment are on opposite sides of the
+                     * boundary, calculate midpoint, the point of intersection
+                     */
+                    outside = positive[iside];
+                    jdim = (idim + 1) & 01; /* the other dimension */
+
+                    midpoint[idim] = border[iside][idim];
+
+                    midpoint[jdim] = (delta[1] * segment[0][jdim] -
+                                      delta[0] * segment[1][jdim]) /
+                                     (delta[1] - delta[0]);
+
+                    if (count == 0) {
+                        /* If a segment cross the boundary the formula for its
+                         * area is a combination of the formulas for segments
+                         * entirely inside and outside.
+                         */
+                        if (outside == 0) {
+                            width = midpoint[0] - segment[0][0];
+                            area += width;
+                            width = segment[1][0] - midpoint[0];
+                            /* Delta[0] is at the crossing point and thus zero
+                             */
+                            area += 0.5 * width * (1.0 + (1.0 + delta[1]));
+                        } else {
+                            width = segment[1][0] - midpoint[0];
+                            area += width;
+                            width = midpoint[0] - segment[0][0];
+                            /* Delta[1] is at the crossing point and thus zero
+                             */
+                            area += 0.5 * width * ((1.0 + delta[0]) + 1.0);
+                        }
 
-          } else {
-            /* Clip segment against each boundary except the last */
-            segment[outside][0] = midpoint[0];
-            segment[outside][1] = midpoint[1];
-          }
+                    } else {
+                        /* Clip segment against each boundary except the last */
+                        segment[outside][0] = midpoint[0];
+                        segment[outside][1] = midpoint[1];
+                    }
+                }
+            }
         }
-      }
-    }
 
-    _nextsegment: continue;
-  }
+    _nextsegment:
+        continue;
+    }
 
-  return fabs(area);
+    return fabs(area);
 }
 
-/** --------------------------------------------------------------------------------------------------
- * Calculate overlap between an arbitrary rectangle, aligned with the axes, and a pixel.
- * This is a simplified version of the compute_area, only valid if axes are nearly aligned.
- * Used by do_kernel_turbo.
+/** ---------------------------------------------------------------------------
```

Comment:
I recommend comments and parameter information in the format shown in a previous comment.